plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/ProgramConverter.kt

```diff
@@ -41,12 +40,34 @@ class ProgramConverter(val session: FirSession, override val config: PluginConfi
             methods = SpecialMethods.all + methods.values.filter { it.shouldIncludeInProgram }.map { it.viperMethod }.toList(),
         )
 
+    fun processRegisteredMethods() {
+        for (method in methodsToVerify) {
+            val embedding = embedFunction(method)
+            embedding.convertBody(this)
+        }
+    }
+
     fun registerForVerification(declaration: FirSimpleFunction) {
```

Comment:
If I see it correctly, then we are only ever calling this method once with the main function being verified. Should `methodToVerify` then even be a list?

Reply:
Eventually we'd like to move towards a situation where we can register many methods, and IME lists tend to be easier to work with than nullables in a case like this.  The alternative is to make `ProgramConverter` take a single method in its constructor, but I think that would be a less nice interface.

Reply:
Actually I would just remove `methodToVerify`, because the current workflow goes something like this:
1. Create `ProgramConverter`
2. Add the one function to the `methodToVerify`
3. Iterate through `methodToVerify` and calling `embedFunction` and `convertBody`
4. Never touch `methodsToVerify` again

So why not just call `embedFunction` and `convertBody` on the one function being verified in the fist place?

Reply:
Hmm, I was thinking that maybe there's some issue of order there, but I think you're right.